src/connection/utils.ts

```diff
@@ -1,14 +1,56 @@
+import BRAVE_ICON from 'assets/wallets/brave-icon.svg'
+import INJECTED_DARK_ICON from 'assets/wallets/browser-wallet-dark.svg'
+import INJECTED_LIGHT_ICON from 'assets/wallets/browser-wallet-light.svg'
+import LEDGER_ICON from 'assets/wallets/ledger-icon.svg'
+import METAMASK_ICON from 'assets/wallets/metamask-icon.svg'
+import PHANTOM_ICON from 'assets/wallets/phantom-icon.svg'
+import RABBY_ICON from 'assets/wallets/rabby-icon.svg'
+import TRUST_WALLET_ICON from 'assets/wallets/trustwallet-icon.svg'
 import { Connection, ConnectionType } from 'connection/types'
 
 export const getIsInjected = () => Boolean(window.ethereum)
 
-// When using Brave browser, `isMetaMask` is set to true when using the built-in wallet
-// This variable should be true only when using the MetaMask extension
-// https://wallet-docs.brave.com/ethereum/wallet-detection#compatability-with-metamask
-type NonMetaMaskFlag = 'isRabby' | 'isBraveWallet' | 'isTrustWallet' | 'isLedgerConnect'
-const allNonMetamaskFlags: NonMetaMaskFlag[] = ['isRabby', 'isBraveWallet', 'isTrustWallet', 'isLedgerConnect']
-export const getIsMetaMaskWallet = () =>
-  Boolean(window.ethereum?.isMetaMask && !allNonMetamaskFlags.some((flag) => window.ethereum?.[flag]))
+const InjectedWalletTable: { [key in keyof NonNullable<Window['ethereum']>]?: { name: string; icon: string } } = {
+  isBraveWallet: { name: 'Brave', icon: BRAVE_ICON },
+  isRabby: { name: 'Rabby', icon: RABBY_ICON },
+  isTrust: { name: 'Trust Wallet', icon: TRUST_WALLET_ICON },
+  isLedgerConnect: { name: 'Ledger', icon: LEDGER_ICON },
+}
+
+/**
+ * Checks the window object for the presence of a known injectors and returns the most relevant injector name and icon.
+ * Returns a default metamask installation object if no wallet is detected.
+ *
+ * @param isDarkMode - optional parameter to determine which color mode of the
+ */
+export function getInjection(isDarkMode?: boolean): { name: string; icon: string } {
+  for (const key in window.ethereum) {
+    const detectedWallet = InjectedWalletTable[key as keyof Window['ethereum']]
+
+    if (detectedWallet) return detectedWallet
+  }
+
+  // Phantom sets it's flag in a different part of the window object
+  if (window.phantom?.ethereum?.isPhantom) return { name: 'Phantom', icon: PHANTOM_ICON }
+
+  // Check for MetaMask last, as other injectors will also set this flag, i.e. Brave browser and Phantom wallet
+  if (window.ethereum?.isMetaMask) return { name: 'MetaMask', icon: METAMASK_ICON }
+
+  // Prompt metamask installation when there is no injection present or the only injection detected is coinbase
```

Comment:
Clarify that we are not attacking coinbase lol

```suggestion
  // Prompt metamask installation when there is no injection present or the only injection detected is coinbase,
  // because coinbase already has its own entry in our UI
```